velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -61,11 +84,13 @@ SubstraitVeloxExprConverter::toVeloxExpr(
   }
   const auto& veloxFunction = substraitParser_.findVeloxFunction(
       functionMap_, substraitFunc.function_reference());
-  const auto& veloxType = toVeloxType(
-      substraitParser_.parseType(substraitFunc.output_type())->type);
-
+  std::string typeName =
+      substraitParser_.parseType(substraitFunc.output_type())->type;
+  if (veloxFunction == VeloxFunNameStrings[VeloxFunName::RowConstructor]) {
+    return toRowConstructorExpr(std::move(params), typeName);
```

Comment:
Why do we need toRowConstructorExpr? Can't we use the following generic code?

```
return std::make_shared<const core::CallTypedExpr>(
      toVeloxType(typeName), std::move(params), veloxFunction);
```

What is the value of "typeName" in case of "row_constructor" function? Why can't toVeloxType convert it into the right type?

Reply:
> Why do we need toRowConstructorExpr? Can't we use the following generic code?
> 
> ```
> return std::make_shared<const core::CallTypedExpr>(
>       toVeloxType(typeName), std::move(params), veloxFunction);
> ```
> 
> What is the value of "typeName" in case of "row_constructor" function? Why can't toVeloxType convert it into the right type?

Hi, @mbasmanova I have a confuse about "toVeloxType" function in TypeUtils.cpp.  The reason why we add hard-code logical "toRowConstructorExpr" before the toVeloxType function is that we thought “::substrait::Type::KindCase::kStruct:” can not responding to TypeKind::ROW directly.  
we thought kStruct may correspond  to TypeKind::kStruct.  Actually,  kStruct is not defined in enum class TypeKind. So If you determine that kStruct is correspond to TypeKind::ROW, I can move the logic of "toRowConstructorExpr" to toVeloxType. And then there is no hard-code.

Reply:
velox::TypeKind::ROW represent a struct type. I assume it would correspond to Substrait's substrait::Type::KindCase::kStruct.